ml/src/test/scala/ai/h2o/sparkling/ml/algos/H2OWord2VecTestSuite.scala

```diff
@@ -0,0 +1,92 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package ai.h2o.sparkling.ml.algos
+
+import ai.h2o.sparkling.{SharedH2OTestContext, TestUtils}
+import org.apache.spark.ml.feature.{RegexTokenizer, StopWordsRemover}
+import org.apache.spark.ml.{Pipeline, PipelineModel}
+import org.apache.spark.sql.SparkSession
+import org.junit.runner.RunWith
+import org.scalatest.junit.JUnitRunner
+import org.scalatest.{FunSuite, Matchers}
+
+@RunWith(classOf[JUnitRunner])
+class H2OWord2VecTestSuite extends FunSuite with Matchers with SharedH2OTestContext {
+
+  override def createSparkSession(): SparkSession = sparkSession("local[*]")
+
+  private lazy val dataset = spark.read
+    .option("header", "true")
+    .option("inferSchema", "true")
+    .csv(TestUtils.locate("smalldata/craigslistJobTitles.csv"))
+
+  private lazy val Array(trainingDataset, testingDataset) = dataset.randomSplit(Array(0.9, 0.1), 42)
+
+  def getPipeline(): Pipeline = {
+    val tokenizer = new RegexTokenizer()
+      .setInputCol("jobtitle")
+      .setMinTokenLength(2)
+
+    val stopWordsRemover = new StopWordsRemover()
+      .setInputCol(tokenizer.getOutputCol)
+
+    val w2v = new H2OWord2Vec()
+      .setSentSampleRate(0)
+      .setEpochs(10)
+      .setFeaturesCol(stopWordsRemover.getOutputCol)
```

Comment:
Shouldn't `H2OWord2Vec` Have `inputCol` and `outputCol` instead of `featuresCol` and `predictionCol`? as it's for `H2OTargetEncoder`?

Reply:
The code is based on H2O where Word 2 Vec is treated as algorithm. I can change though, don't have really strong preference

Reply:
Removing featuresCol and predictionCol would mean that changes to the structure of H2OALgorithm classes etc as these parameters are defined there ( and Word2Vec depends on H2OAlgorithm)

I can change it, but would first merge PR in this state and iterate after in another PR

Reply:
I have exposed inputCol and outputCol and up-dated also the documentation. But still kept also params featuresCol and predictionCol available. Those can be removed in the follow-up PR, want to avoid doing that as part of this change.

Reply:
because it shares all the pieces with H2OAlgorithm, mojo etc. Just prevents duplication the code.

I'm open to changing the structure as explained above, but I'm in favor of keeping it child-class of H2OAlgorithm ( since it is algorithm in H2O)